packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -248,22 +249,28 @@ const helpers = {
     return helpers.negotiateOptionsForDefaultCode(httpOperation, desiredOptions);
   },
 
-  findResponse(httpResponses: IHttpOperationResponse[]): Reader.Reader<Logger, Option.Option<IHttpOperationResponse>> {
+  findResponse(
+    httpResponses: IHttpOperationResponse[],
+    statusCodes: NonEmptyArray<string>,
+  ): Reader.Reader<Logger, Option.Option<IHttpOperationResponse>> {
+    const [first, ...others] = statusCodes;
+
     return withLogger<Option.Option<IHttpOperationResponse>>(logger =>
       pipe(
-        findResponseByStatusCode(httpResponses, '422'),
-        Option.alt(() => {
-          logger.trace('Unable to find a 422 response definition');
-          return findResponseByStatusCode(httpResponses, '400');
-        }),
-        Option.alt(() => {
-          logger.trace('Unable to find a 400 response definition.');
-          return findResponseByStatusCode(httpResponses, '401');
-        }),
-        Option.alt(() => findResponseByStatusCode(httpResponses, '403')),
+        others.reduce(
+          (previous, current, index) =>
+            pipe(
+              previous,
+              Option.alt(() => {
+                logger.trace(`Unable to find a ${statusCodes[index === 0 ? first : 1 + index]} response definition`);
```

Comment:
@XVincentX,
if `statusCodes` equals ['400', '422', '404'], then
for `index === 0`: statusCodes['400'],
for `index === 1`: statusCodes['2']
for `index === 2`: statusCodes['3']

or am I reading this wrong?

Reply:
That's effectively wrong, I've fixed it.